lib/pliny/librato/metrics/backend.rb

```diff
@@ -24,61 +24,66 @@ def report_measures(measures)
         end
 
         def start
-          start_thread
+          start_counter
+          start_timer
           self
         end
 
         def stop
           metrics_queue.push(POISON_PILL)
-          thread.join
+          timer.terminate
```

Comment:
If this runs while the thread has the mutex, it will not return it, causing `flush_librato` to deadlock. This is where this gets hard, the first thing that came to mind is to use `Thread#join` and use a `boolean` to indicate the shut down, but the `sleep` won't resume until it's time is up, blocking the execution.

The better option I think is to obtain the mutex, and `Thread#terminate` on both threads. This ensures that neither thread is inside the critical section. The thing to be careful about there is that `flush_librato` has to happen after the mutex is returned, since Ruby mutexes are not re-entrant.